echopype/convert/set_groups_ek80.py

```diff
@@ -1310,32 +1317,93 @@ def set_vendor(self) -> xr.Dataset:
         if "impedance" in ds_cal:
             ds_cal = ds_cal.rename_vars({"impedance": "impedance_transducer"})
 
-        #  Save decimation factors and filter coefficients
-        coeffs = dict()
-        decimation_factors = dict()
+        # Save decimation factors and filter coefficients
+        # Param map values
+        # 1: wide band transceiver (WBT)
+        # 2: pulse compression (PC)
+        param_map = {1: WIDE_BAND_TRANS, 2: PULSE_COMPRESS}
+        coeffs_and_decimation = {
+            t: {FILTER_IMAG: [], FILTER_REAL: [], DECIMATION: []} for t in list(param_map.values())
+        }
+
         for ch in self.sorted_channel["all"]:
-            # filter coeffs and decimation factor for wide band transceiver (WBT)
-            if self.parser_obj.fil_coeffs and (ch in self.parser_obj.fil_coeffs.keys()):
-                coeffs[f"{ch} WBT filter"] = self.parser_obj.fil_coeffs[ch][1]
-                decimation_factors[f"{ch} WBT decimation"] = self.parser_obj.fil_df[ch][1]
-            # filter coeffs and decimation factor for pulse compression (PC)
-            if self.parser_obj.fil_df and (ch in self.parser_obj.fil_coeffs.keys()):
-                coeffs[f"{ch} PC filter"] = self.parser_obj.fil_coeffs[ch][2]
-                decimation_factors[f"{ch} PC decimation"] = self.parser_obj.fil_df[ch][2]
+            fil_coeffs = self.parser_obj.fil_coeffs.get(ch, None)
+            fil_df = self.parser_obj.fil_df.get(ch, None)
+
+            if fil_coeffs and fil_df:
+                # get filter coefficient values
+                for type_num, values in fil_coeffs.items():
+                    param = param_map[type_num]
+                    coeffs_and_decimation[param][FILTER_IMAG].append(np.imag(values))
+                    coeffs_and_decimation[param][FILTER_REAL].append(np.real(values))
+
+                # get decimation factor values
+                for type_num, value in fil_df.items():
+                    param = param_map[type_num]
+                    coeffs_and_decimation[param][DECIMATION].append(value)
```

Comment:
Cleaned this up, let me know what you think @leewujung. Thanks!

Reply:
`param = param_map[type_num]`
I like this! :D